man/systemd-sysusers.xml

```diff
@@ -94,6 +94,22 @@
         paths. </para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--config-name=<replaceable>NAME</replaceable></option></term>
```

Comment:
so, i'd not add this "--config-name=" switch here, but always read the full data instead. Why? For three reasons:

1. It makes things simpler, and removes the necessity for this new argument (in particular one with an ugly abbreviation such as "config"... ;-))
2. We say in the docs that sysusers.d is "idempotent", i.e. that calling it many times has the same result as calling it once. And I think we should stay as true as possible to that. However, if the invocation from the RPM scriptlets operates on a very different set of rules than the usual invocations then this idempotency is kinda out of the window, no? Yes, strictly speaking, regardless what we do, if we accept data from the cmdline in addition to what's on disk, we aren't idempotent in this regard anyway anymore, since it's means the invocations are only identical if called with the very same non-trivial arguments, but I think this is OK, since in the intended usecase we'll add the right file to disk one instants later anyway, hence the idempotency is restored...
3. I think there are strong usecases to permit admits to place /etc/sysusers.d/00-mycompany.conf or so, and have that define fixed UID mappings the admin wants, but for that to work correctly, every invocation of sysusers should really take that into account...

Reply:
OK. I updated the code to read everything in /etc and /run when command line arguments are given. This way we preserve the requirements you describe, but we don't have to read and execute all of configuration, which can be quite large.